config/config.go

```diff
@@ -0,0 +1,75 @@
+package config
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	DSN      string
+	InfoLog  *log.Logger
+	ErrorLog *log.Logger
+	DB       *sql.DB
+	Addr     *string
+}
+
+// Init the application's configuration
+func Init(infoLog, errorLog *log.Logger) (*Config, error) {
+
+	//Read config file
+	viper.SetConfigName("config")
+	viper.SetConfigType("json")
+	viper.AddConfigPath(".")
+	err := viper.ReadInConfig()
+
+	if err := viper.ReadInConfig(); err != nil {
+		if _, ok := err.(viper.ConfigFileNotFoundError); ok {
+			// Config file not found
+			panic(fmt.Errorf("fatal error config file not found: %s", err))
+
+		} else {
+			// Config file was found but another error was produced
+			panic(fmt.Errorf("fatal error config file: %s", err))
+		}
+	}
+
+	// Get the server address/port
+	addr := viper.GetString("addr")
+
+	// Create the Data Source Name
+	// *We need to use the parseTime=true parameter in our
+	// DSN to force it to convert TIME and DATE fields to time.Time. Otherwise it returns these as
+	// []byte objects.
+	username := viper.GetString("db.username")
+	password := viper.GetString("db.password")
+	dbHost := viper.GetString("db.host")
+	dbPort := viper.GetString("db.port")
+	dbName := viper.GetString("db.name")
+	dsn := fmt.Sprintf("%s:%s@(%s:%s)/%s?parseTime=true", username, password, dbHost, dbPort, dbName)
+
+	// Open db connection
+	db, err := openDB(dsn)
+	if err != nil {
+		errorLog.Fatalf("message: unable to open db connection, type: database, err: %v", err)
+		return nil, err
+	}
+
+	// Retrun the config
+	return &Config{dsn, infoLog, errorLog, db, &addr}, nil
```

Comment:
If you agree with the previous comment, please remove `db` from the Config struct, as that's not the right place for it.